### PR TITLE
fix(rcm): add git metadata to requests

### PR DESCRIPTION
**Pull request by P403n1x87**
We make sure that git metadata is added to the tags when making requests for configuration to the agent. This allows SCI to work as expected with services that rely on the git metadata to be transmitted with RC requests.

We also add the tags to every DI upload.

## Testing strategy

This will be covered by system tests.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by mabdinur — COMMENTED** [review]
Can we add a test case. Aside from that LGTM

File: ddtrace/internal/gitmetadata.py [hunk 1]
```diff
@@ -124,3 +124,15 @@ def clean_tags(tags):
     tags.pop(COMMIT_SHA, None)
 
     return tags
+
+
+def update_tags(tags):
```
Comment (shatzi) [inline ddtrace/internal/gitmetadata.py:129]: would rename method to indicate better what it does.
for example: add_git_tags(tags).

**Review by shatzi — APPROVED** [review]
Do we want to ensure git metadata tags are also in snapshots messages/ probe status as well?

**Comment by P403n1x87** [comment]
> Do we want to ensure git metadata tags are also in snapshots messages/ probe status as well?

Is that needed for SCI to work? If so then we should definitely add them there too.

**Review by majorgreys — APPROVED** [review]

**Review by shatzi — APPROVED** [review]
looks good.

**Pull request merged**